kernel/timeout.c

```diff
@@ -73,7 +73,8 @@ static int32_t next_timeout(void)
 {
 	struct _timeout *to = first();
 	int32_t ticks_elapsed = elapsed();
-	int32_t ret = to == NULL ? MAX_WAIT : MAX(0, to->dticks - ticks_elapsed);
+	int32_t ret = to == NULL ? MAX_WAIT :
+		MAX(0, (int32_t)(to->dticks - ticks_elapsed));
```

Comment:
This change would be incorrect.  Since #28034 `to->dticks` is either `int32_t` or `int64_t`.  `ticks_elapsed` is `int32_t`.  If `to->dticks` is `int64_t` then the subtraction will produce an `int64_t` value, and casting that to `int32_t` will produce an implementation-defined result (possibly raising a signal) if the value cannot be exactly represented.  (ISO/IEC 9899 2018 6.3.1.3).

Using signed arithmetic I believe this needs to be:
```
int32_t ret = to == NULL ? MAX_WAIT :
￼		MIN(MAX_WAIT, MAX(0, (to->dticks - ticks_elapsed)));
```

Reply:
It looks right, though this is now tackling a different issue (making sure int64_t is not cast to int32_t). I'll go ahead and close this PR. Would you like to submit a new one with the change?

Reply:
Yes, I'll do that.